### PR TITLE
feat: add workspace id and authz to shells

### PR DESCRIPTION
**Pull request by eecsliu**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: proto/src/determined/api/v1/shell.proto [hunk 1]
```diff
@@ -37,6 +37,9 @@ message GetShellsRequest {
   repeated string users = 5;
```
Comment (hamidzr) [inline proto/src/determined/api/v1/shell.proto:37]: comparing with https://github.com/determined-ai/determined/pull/5730/files?show-viewed-files=true&file-filters%5B%5D= looks like you're missing the sort parameter

File: master/internal/api_shell.go [hunk 2]
```diff
@@ -51,18 +52,16 @@ func (a *apiServer) GetShells(
 	if err = a.ask(shellsAddr, req, &resp); err != nil {
 		return nil, err
 	}
-
+	limitedScopes, err := command.AuthZProvider.Get().AccessibleScopes(
+		ctx, *curUser, model.AccessScopeID(req.WorkspaceId),
+	)
+	if err != nil {
+		return nil, apiutils.MapAndFilterErrors(err, nil, nil)
```
Comment (hamidzr) [inline master/internal/api_shell.go:59]: - i don't have an intg test for the plural version of getNTSC
- are there cases we'd wanna swap the permission erro for 404?

File: master/internal/api_shell.go [hunk 3]
```diff
@@ -106,7 +117,19 @@ func (a *apiServer) KillShell(
...
+	}
+
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = command.AuthZProvider.Get().CanSetNSCsPriority(
+		ctx, *curUser, model.AccessScopeID(getResponse.Shell.WorkspaceId), int(req.Priority))
+	if err != nil {
```
Comment (hamidzr) [inline master/internal/api_shell.go:132]: these can leak that the entity exists could you hook it up to tests here https://github.com/hamidzr/determined/blob/b6f7f5ea0005a9a2e39fa6a0a6c3c9fdb3f810fb/master/internal/api_ntsc_intg_test.go#L176 ? 

although looking at it again the assertion in the test probably has to be more complicated based on if the user has view access or not. but defaulting to 404 seems like a saner option?

File: master/internal/api_shell.go [hunk 4]
```diff
@@ -125,6 +153,17 @@ func (a *apiServer) LaunchShell(
 		return nil, api.APIErrToGRPC(errors.Wrapf(err, "failed to prepare launch params"))
 	}
 
+	workspaceID := model.DefaultWorkspaceID
```
Comment (hamidzr) [inline master/internal/api_shell.go:156]: did you wanna use `isNTSCPermittedToLaunch` ?

**Review by hamidzr — COMMENTED** [review]
added some comments. and compared the changes w/ how we're adding [commands here](https://github.com/determined-ai/determined/pull/5730/files?show-viewed-files=true&file-filters%5B%5D=)
it looks like we need to be more consistent with the errors types and ensuring that information isn't leaked. I'll have to update some of the tests for that as well

File: master/internal/api_shell.go (hunk 4 above)
Comment (eecsliu) [reply master/internal/api_shell.go]: just added!

File: master/internal/api_shell.go (hunk 3 above)
Comment (eecsliu) [reply master/internal/api_shell.go:132]: sg, I'll add tests and check out the response. I have yet to modify things into 404 errors.

File: proto/src/determined/api/v1/shell.proto (hunk 1 above)
Comment (eecsliu) [reply proto/src/determined/api/v1/shell.proto:37]: ah good catch

**Review by hamidzr — APPROVED** [review]
looks great. I think we could have included the cli changes here as well but let's leave that for a separate pr and land this sooner.

**Pull request merged**